src/net/server/dgram.rs

```diff
@@ -502,74 +509,28 @@ where
-                                let mut stream = svc.call(request).await;
-                                while let Some(Ok(call_result)) = stream.next().await {
-                                    let (response, feedback) = call_result.into_inner();
-
-                                    if let Some(feedback) = feedback {
-                                        match feedback {
-                                            ServiceFeedback::Reconfigure {
-                                                idle_timeout: _, // N/A - only applies to connection-oriented transports
-                                            } => {
-                                                // Nothing to do.
-                                            }
-
-                                            ServiceFeedback::BeginTransaction|ServiceFeedback::EndTransaction => {
-                                                // Nothing to do.
-                                            }
-                                        }
-                                    }
-
-                                    // Process the DNS response message, if any.
-                                    if let Some(response) = response {
-                                        // Convert the DNS response message into bytes.
-                                        let target = response.finish();
-                                        let bytes = target.as_dgram_slice();
-
-                                        // Logging
-                                        if enabled!(Level::TRACE) {
-                                            let pcap_text = to_pcap_text(bytes, bytes.len());
-                                            trace!(%addr, pcap_text, "Sending response");
-                                        }
-
-                                        metrics.inc_num_pending_writes();
-
-                                        // Actually write the DNS response message bytes to the UDP
-                                        // socket.
-                                        if let Err(err) = Self::send_to(
-                                            &cloned_sock,
-                                            bytes,
-                                            &addr,
-                                            write_timeout,
-                                        )
-                                        .await
-                                        {
-                                            warn!(%addr, "Failed to send response: {err}");
-                                        }
-
-                                        metrics.dec_num_pending_writes();
-                                        metrics.inc_num_sent_responses();
-                                    }
-                                }
-                            }
+                    match Message::from_octets(buf) {
+                        Err(err) => {
```

Comment:
It seems that in this case no error gets returned to the client

Reply:
We don't have a DNS request from which to extract a query ID to copy into the error response. So we could generate a FORMERR response and send it, but with what ID?

Reply:
Note: When merged with `main` there is an additional `Ok(msg) if msg.header().qr() => ` match stanza which also doesn't send a reply, but that DOES know the ID of the request and so should respond.